### ISSUE TITLE
Regression in markdown reader: bracketed spans and latex macros

### ISSUE THREAD

**Issue by dsteurer** [jgm/pandoc#4530]
For the following `markdown` input,
````
\newcommand{\set}[1]{\{#1\}}
[$0 \in \set{0,1}$]{.center}
````
the latest release `pandoc 2.1.3` doesn't parse the bracketed span.
For example, the command `pandoc -f markdown -t html5 --mathjax` (with the above as standard input) produces the following output
````
<p>[<span class="math inline">\(0 \in \{0,1\}\)</span>]{.center}</p>
````
A previous release `pandoc 2.0.2` produces the expected output:
````
<p><span class="center"><span class="math inline">\(0 \in \{0,1\}\)</span></span></p>
````
The reason appears to be the latex macro `\set` because the latest release `pandoc 2.1.3` produces the expected output for the following `markdown` input:
````
[$0 \in \{0,1\}$]{.center}
````
There is also some interaction with the latex command `\in` because `pandoc 2.1.3` produces the expected output for this input:
````
\newcommand{\set}[1]{\{#1\}}
[$0 {\in} \set{0,1}$]{.center}
````
Here is the output of `pandoc -v`:
````
pandoc 2.1.3
Compiled with pandoc-types 1.17.4.2, texmath 0.10.1.1, skylighting 0.7.0.2
Default user data directory: /home/david/.pandoc
Copyright (C) 2006-2018 John MacFarlane
Web:  http://pandoc.org
This is free software; see the source for copying conditions.
There is no warranty, not even for merchantability or fitness
for a particular purpose.
````

**Comment by jgm** [comment]
I'm guessing this might be due to commit d9584d73f94501787026c57b77d217e51f21505d

Can you try with 2.1 and 2.1.1 to confirm this guess?

**Comment by dsteurer** [comment]
Indeed, 2.1 gives the expected output and 2.1.1 gives the same output as the latest release.

**Comment by jgm** [comment]
This seems to be fixed by 06bc0376d1cc26b254e9f85a5dd7e29c4e969c3c and 06bc0376d1cc26b254e9f85a5dd7e29c4e969c3c.
